packages/manager/apps/dedicated/client/app/network-security/traffic/traffic.controller.js

```diff
@@ -0,0 +1,282 @@
+import get from 'lodash/get';
+import map from 'lodash/map';
+
+import {
+  TRAFFIC_PERIODS,
+  TRAFFIC_PERIOD_LIST,
+  CHART,
+} from './traffic.constant';
+
+export default class TrafficController {
+  /* @ngInject */
+  constructor($http, $state, $translate, Alerter, networkSecurityService) {
+    this.$http = $http;
+    this.$state = $state;
+    this.$translate = $translate;
+    this.Alerter = Alerter;
+    this.networkSecurityService = networkSecurityService;
+    this.TRAFFIC_PERIODS = TRAFFIC_PERIODS;
+    this.TRAFFIC_PERIOD_LIST = TRAFFIC_PERIOD_LIST;
+    this.CHART = CHART;
+  }
+
+  $onInit() {
+    this.errorMessage = '';
+    this.pageSize = 10;
+    this.isStackable = true;
+    this.isPPs = false;
+    this.subnet = '';
+    this.periods = this.networkSecurityService.initPeriods(
+      this.TRAFFIC_PERIODS,
+    );
+    [this.period] = this.periods;
+    this.networkSecurityService.initService().then((data) => {
+      this.services = data;
+      return data;
+    });
+
+    this.units = this.CHART.units;
+    this.colors = this.CHART.colors;
+    this.options = {
+      ...this.CHART.options,
+    };
+  }
+
+  selectService() {
+    if (this.service) {
+      this.pageSize = 10;
+      this.page = 1;
+      this.autocomplete = [];
+      this.selectedIp = '';
+      this.results = null;
+      this.networkSecurityService
+        .getIpsFromService(
+          this.page,
+          this.pageSize,
+          this.service.serviceName,
+          this.autocomplete,
+        )
+        .then((data) => {
+          this.autocomplete = data;
+        });
+    }
+  }
+
+  getIps(partial) {
+    if (!this.autocomplete) {
+      return null;
+    }
+    let ips = [];
+    if (partial.length > 2) {
+      this.loaderIp = true;
+
+      // Filter loaded ips list with partial ip
+      ips = this.autocomplete.filter((service) =>
+        service.ipBlock.includes(partial),
+      );
+      delete this.loaderIp;
+    }
+    return ips;
+  }
+
+  checkSelectedSubnet(value) {
+    if (!value) {
+      return;
+    }
+    this.subnet = value.ipBlock ? value.ipBlock : value;
+    this.getTraffic();
+  }
+
+  getTraffic() {
+    if (this.$state.params.subnet) {
+      this.subnet = this.$state.params.subnet;
```

Comment:
```suggestion
      this.subnet = this.getSubnet();
```
retrieve subnet in routing resolve so you don't need to make $state accessible in the scope

Reply:
I would like but how I can get this value into the controller from the routing?